pkg/armhelpers/groupsclient.go

```diff
@@ -18,3 +20,16 @@ func (az *AzureClient) EnsureResourceGroup(name, location string) (resourceGroup
 
 	return &response, nil
 }
+
+// DeleteResourceGroup delete the named resource group
+func (az *AzureClient) DeleteResourceGroup(name string, cancel chan struct{}) error {
+	resCh, errCh := az.groupsClient.Delete(name, cancel)
+	res := <-resCh
+	err := <-errCh
+	// When resourceGroup not exists, we will get 404
+	// Explictly set the error to nil for this scenario
+	if res.Response != nil && res.StatusCode == http.StatusNotFound {
```

Comment:
I have removed this part. It is our RP's logic. 

res.StatusCode is actually res.Response.Response.StatusCode. It is the embedding confusion.